pkg/apis/core/helper/helpers.go

```diff
@@ -146,10 +148,25 @@ func IsStandardContainerResourceName(str string) bool {
 	return standardContainerResources.Has(str) || IsHugePageResourceName(core.ResourceName(str))
 }
 
-// IsExtendedResourceName returns true if the resource name is not in the
-// default namespace.
+// IsExtendedResourceName returns true if:
+// 1. the resource name is not in the default namespace;
+// 2. its length not exceed extendedResourceNameMaxLength(54),
+// to ensure quota for the resource(requests.[resourceName])
+// not exceed qualifiedNameMaxLength(63)
+// 3. resource name does not have "requests." prefix,
+// to avoid confusion with the convention in quota
 func IsExtendedResourceName(name core.ResourceName) bool {
-	return !IsDefaultNamespaceResource(name)
+	return !IsDefaultNamespaceResource(name) &&
+		len(string(name)) <= extendedResourceNameMaxLength &&
+		!strings.HasPrefix(string(name), core.DefaultResourceRequestsPrefix)
+}
+
+// IsExtendedResourceNameForQuota returns true if the extended resource name
+// has the quota related resource prefix.
+func IsExtendedResourceNameForQuota(name core.ResourceName) bool {
```

Comment:
Do we still need to define this function here? Can we instead define it as a local func in pkg/quota/evaluator/core/pods.go?